pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -409,6 +393,34 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	return err
 }
 
+// updateNodePodCIDRWithCidrStrings update the Node object with Spec.PodCIDR(s),
+// returns error if cidrStrings is not valid or fails to update the Node object
+func (ca *cloudCIDRAllocator) updateNodePodCIDRWithCidrStrings(oldNode *v1.Node, node *v1.Node, cidrStrings []string) error {
+	if len(cidrStrings) == 0 {
+		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
+		return fmt.Errorf("failed to allocate cidr: Node %v has no CIDRs", node.Name)
+	}
+	// Can have at most 2 ips (one for v4 and one for v6)
+	if len(cidrStrings) > 2 {
+		klog.InfoS("Got more than 2 ips, truncating to 2", "cidrStrings", cidrStrings)
+		cidrStrings = cidrStrings[:2]
+	}
```

Comment:
@sdmodi I want to run this for you, I think I remember one issue that the cidrstrings where [IPv4, IPv4, IPv6] , I can't find it right now, it was with one special setup used by scalability folks, how do you think this should work?

If it truncates to IPv4,IPv4 this will fail later when checking for dualstackness
What we ended doing in the kubelet is to parse the whole array and take the first IPv4 and the first IPv6, 

I don't have clear what is best, I don't feel strongly to any of those options

Reply:
I didn't think there was a legitimate case where you would see more than 2 CIDR strings. If we are truncating this, a better logic would be to pick the first IPv4 and first Ipv6 addresses. I don't have a strong opinion on whether we should do this or not since this is an unusual case.

Reply:
agree, let's keep the existing behavior then